sql/core/src/main/scala/org/apache/spark/sql/execution/columnar/compression/compressionSchemes.scala

```diff
@@ -421,7 +421,7 @@ private[columnar] case object DictionaryEncoding extends CompressionScheme {
 
     override def compress(from: ByteBuffer, to: ByteBuffer): ByteBuffer = {
       if (overflow) {
```

Comment:
If users never face to it, let's replace it by an assert. But I guess users might face to the restriction:
```scala
  // 32K unique values allowed
  val MAX_DICT_SIZE = Short.MaxValue
```
@goutam-git Could you try to create big enough dictionary?

Reply:
@MaxGekk

    I used the DictionaryEncodingSuite and found that in case of big enough Dictionary the users face java.lang.IllegalArgumentException  from ByteBuffer.allocate in build method in CompressibleColumnBuilder as a result of which they never reach encoder.compress  which throws  useDictionaryEncodingWhenDictionaryOverflowError
Please advice next steps

Reply:
Then, please, replace `if (overflow) {` by an assert with a comment which explains why the condition cannot be false. Also don't forget to remove `useDictionaryEncodingWhenDictionaryOverflowError()`.

Reply:
@MaxGekk

on  replacing if(overflow) with asser ,  pyspark test for CrossValidatorTests is failing for this assertion.

 

File "/__w/spark/spark/python/pyspark/ml/tests/test_tuning.py", line 141, in test_copy

    cvModel = cv.fit(dataset)

 

py4j.protocol.Py4JJavaError: An error occurred while calling o48.evaluate.

: org.apache.spark.SparkException: Job aborted due to stage failure: Task 0 in stage 3.0 failed 1 times, most recent failure: Lost task 0.0 in stage 3.0 (TID 12) (localhost executor driver): java.lang.AssertionError: assertion failed

               at scala.Predef$.assert(Predef.scala:208)

               at org.apache.spark.sql.execution.columnar.compression.DictionaryEncoding$Encoder.compress(compressionSchemes.scala:425)

 
Please advice.

Reply:
@MaxGekk  should I remove the assert and use the new method name with error class instead of useDictionaryEncodingWhenDictionaryOverflowError()